### ISSUE TITLE
Better error message for incoming webhooks without mime type

### ISSUE THREAD

**Issue by N-Schaef** [mattermost/mattermost-server#14485]
Hello,
since updating the server, I noticed that my webhooks are not working anymore.
This seems to be the case, because the check against mime types was slightly changed:

https://github.com/mattermost/mattermost-server/blob/10538b481357f186949a082afef9999b01ab982e/web/webhook.go#L33

The error came from a library, which does not set the `Content-Type` header in it's request.
This is of course wrong and I opened an issue there.

But the error message that will be returned is
```json
{"id":"api.webhook.incoming.error","message":"Could not decode the multipart payload of incoming webhook.","detailed_error":"","request_id":"<id>","status_code":400}
```
which I find a little bit cryptic.
Maybe it would be helpful to prevent future errors/support requests if there would be a specific "missing/unsupported Content-Type header".

**Comment by amyblais** [comment]
@N-Schaef Are you on server v5.22? What OS and version are you on?

**Comment by N-Schaef** [comment]
Yes 5.22.1 on debian buster.

But as I said, I guess this is expected behaviour and is already being fixed in the library.
The slack API behaves in exactly the same way.

I just wanted to suggest to check against missing Content-Type headers and then writing a note in the `detailed_error` section of the response to give better feedback to the user

**Comment by amyblais** [comment]
Opened a ticket for this here: https://mattermost.atlassian.net/browse/MM-24977.

**Comment by amyblais** [comment]
Was fixed in v5.23.1 dot release,
